zone/bot.cpp

```diff
@@ -2449,1144 +1955,989 @@ void Bot::AI_Process()
-							DualWieldProbability += (DualWieldProbability * float(DWBonus) / 100.0f);
+			else {
+				melee_distance = melee_distance_max * 0.25f;
+			}
+			break;
+		}
+			// Fall-through
+	default:
+		if (p_item && p_item->GetItem()->IsType2HWeapon()) {
+			melee_distance = melee_distance_max * 0.70f;
+		}
+		else {
+			melee_distance = melee_distance_max * 0.50f;
+		}
 
-							float random = zone->random.Real(0, 1);
-							if (random < DualWieldProbability) { // Max 78% of DW
+		break;
+	}
+}
 
-								Attack(tar, EQ::invslot::slotSecondary);	// Single attack with offhand
+bool Bot::IsValidTarget(Client* bot_owner, Client* leash_owner, float lo_distance, float leash_distance, bool bo_alt_combat, Mob* tar, float tar_distance) {
 
-								TEST_COMBATANTS();
-								TryCombatProcs(s_item, tar, EQ::invslot::slotSecondary);
+	if (HOLDING ||
+		!tar->IsNPC() ||
+		tar->IsMezzed() ||
+		lo_distance > leash_distance ||
+		tar_distance > leash_distance ||
+		(!GetAttackingFlag() && !CheckLosFN(tar) && !leash_owner->CheckLosFN(tar)) || // This is suppose to keep bots from attacking things behind walls
+		!IsAttackAllowed(tar) ||
+		(bo_alt_combat &&
+			(!GetAttackingFlag() && NOT_PULLING_BOT && !leash_owner->AutoAttackEnabled() && !tar->GetHateAmount(this) && !tar->GetHateAmount(leash_owner))
+		)
+	)
+	{
+		// Normally, we wouldn't want to do this without class checks..but, too many issues can arise if we let enchanter animation pets run rampant
+		if (HasPet()) {
+			GetPet()->RemoveFromHateList(tar);
+			GetPet()->SetTarget(nullptr);
+		}
 
-								TEST_COMBATANTS();
-								if (CanThisClassDoubleAttack() && CheckBotDoubleAttack()) {
+		RemoveFromHateList(tar);
+		SetTarget(nullptr);
 
-									if (tar->GetHP() > -10) {
-										Attack(tar, EQ::invslot::slotSecondary);	// Single attack with offhand
-									}
-								}
-							}
-						}
-					}
-				}
-			}
+		SetAttackFlag(false);
+		SetAttackingFlag(false);
 
-			if (GetAppearance() == eaDead) {
-				return;
+		if (PULLING_BOT) {
+			SetPullingFlag(false);
+			SetReturningFlag(false);
+			bot_owner->SetBotPulling(false);
+			if (GetPet()) {
+				GetPet()->SetPetOrder(m_previous_pet_order);
 			}
 		}
 
-//#pragma endregion
+		if (IsMoving()) {
+			StopMoving();
+		}
 
-//#pragma region ENGAGED NOT AT COMBAT RANGE
+		return false;
+	}
+	return true;
+}
 
-		else { // To far away to fight (GetTarget() validity can be iffy below this point - including outer scopes)
+Mob* Bot::GetBotTarget(Client* bot_owner) {
 
-			// This code actually gets processed when we are too far away from target and have not engaged yet, too
-			if (/*!GetCombatJitterFlag() && */AI_movement_timer->Check() && (!spellend_timer.Enabled() || GetClass() == BARD)) { // Pursue processing
+	Mob* tar = GetTarget();
+	if (!tar || PASSIVE) {
+		if (GetTarget()) {
+			SetTarget(nullptr);
+		}
 
-				if (GetTarget() && !IsRooted()) {
+		WipeHateList();
```

Comment:
Lots of formatting changes in this file that makes it difficult to review

Reply:
Apologies, I needed to go through and refactor a large portion of bot.cpp regarding the Bot AI Process & Spell Processes to incorporate any Raid checks, without doing so it would have been a nightmare imo.

The alternative would have been to keep the duplicated Raid versions of bot.cpp and botspellsai.cpp which I just couldn't stomach :(